packages/openapi-code-generator/src/typescript/typescript-koa/typescript-koa.generator.ts

```diff
@@ -30,96 +30,138 @@ function reduceParamsToOpenApiSchema(parameters: IRParameter[]): IRModelObject {
     const models = this.models
     const schemaBuilder = this.schemaBuilder
 
     const pathParams = operation.parameters.filter(it => it.in === "path")
-
     const paramSchema = pathParams.length ? schemaBuilder.fromParameters(pathParams) : undefined
     let pathParamsType = "void"
 
     const queryParams = operation.parameters.filter(it => it.in === "query")
     const querySchema = queryParams.length ? schemaBuilder.fromParameters(queryParams) : undefined
     let queryParamsType = "void"
 
-    const { requestBodyParameter } = requestBodyAsParameter(operation)
+    const {requestBodyParameter} = requestBodyAsParameter(operation)
     const bodyParamIsRequired = Boolean(requestBodyParameter?.required)
     const bodyParamSchema = requestBodyParameter ? schemaBuilder.fromModel(requestBodyParameter.schema, requestBodyParameter.required) : undefined
     let bodyParamsType = "void"
 
     if (paramSchema) {
-      const name = `${ operation.operationId }ParamSchema`
-      pathParamsType = models.schemaObjectToType({ $ref: this.input.loader.addVirtualType(operation.operationId, _.upperFirst(name), reduceParamsToOpenApiSchema(pathParams)) })
-      this.statements.push(`const ${ name } = ${ paramSchema.toString() }`)
+      const name = `${operation.operationId}ParamSchema`
+      pathParamsType = models.schemaObjectToType({$ref: this.input.loader.addVirtualType(operation.operationId, _.upperFirst(name), reduceParamsToOpenApiSchema(pathParams))})
+      this.statements.push(`const ${name} = ${paramSchema.toString()}`)
     }
 
     if (querySchema) {
-      const name = `${ operation.operationId }QuerySchema`
+      const name = `${operation.operationId}QuerySchema`
       queryParamsType = models.schemaObjectToType({
         $ref: this.input.loader.addVirtualType(operation.operationId, _.upperFirst(name), reduceParamsToOpenApiSchema(queryParams)),
       })
-      this.statements.push(`const ${ name } = ${ querySchema.toString() }`)
+      this.statements.push(`const ${name} = ${querySchema.toString()}`)
     }
 
     if (bodyParamSchema && requestBodyParameter) {
-      const name = `${ operation.operationId }BodySchema`
+      const name = `${operation.operationId}BodySchema`
       bodyParamsType = models.schemaObjectToType({
         $ref: this.input.loader.addVirtualType(operation.operationId, _.upperFirst(name), this.input.schema(requestBodyParameter.schema)),
       })
-      this.statements.push(`const ${ name } = ${ bodyParamSchema }`)
+      this.statements.push(`const ${name} = ${bodyParamSchema}`)
+    }
+
+    const statusStringToType = (status: string) => {
+      if (/^\d+$/.test(status)) {
+        return status
+      } else if (/^\d[xX]{2}$/.test(status)) {
+        return `StatusCode${status[0]}xx`
+      }
+      throw new Error(`unexpected status string '${status}'`)
     }
 
+    const responseSchemas = Object.entries(operation.responses ?? {}).reduce((acc, [status, response]) => {
```

Comment:
This feels like it should be extracted into it's own thing, but I think this will happen as part of supporting multiple response content types